var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -192,18 +192,21 @@ def makefile(self):
         makefile_basename = '.'.join([
             self.makefile_architecture, self.makefile_version
         ])
-        return os.path.join('arch', makefile_basename)
+        return join_path('arch', makefile_basename)
 
     @property
     def archive_files(self):
-        return [os.path.join(self.stage.source_path, self.makefile)]
+        return [join_path(self.stage.source_path, self.makefile)]
 
     def edit(self, spec, prefix):
         pkgconf = which('pkg-config')
 
         if '^fftw' in spec:
             fftw = spec['fftw:openmp' if '+openmp' in spec else 'fftw']
             fftw_header_dir = fftw.headers.directories[0]
+        elif '^amdfftw' in spec:
+            fftw = spec['amdfftw:openmp' if '+openmp' in spec else 'amdfftw']
```

Comment:
this probably requires a `conflicts('^amdfftw~openmp', when='+openmp')`(similar to fftw) to avoid silent/strange errors should the user not have `amdfftw+openmp` right from the start

Reply:
Agree, addressed the same.

Reply:
thanks!